rpc/server.go

```diff
@@ -120,6 +121,18 @@ func (s *Server) readWrite(ret chan<- error) {
 			break
 		} else if err != nil {
 			logrus.Errorf("Failed to read: %v", err)
+			// check if client is valid, since MagicVersion will always
+			// be a constant for valid clients such as jiva controller.
+			// This is done to ignore the requests from invalid clients
+			// such as Prometheus which by default scraps from all the open
+			// ports.
+			msg, errRead := s.wire.ReadMessage()
```

Comment:
I don't understand the need for this code, can you please explain a bit more over here. There already is a check in s.wire.Read() which verifies the same.

Reply:
yes, but there it will return the error and fatal the replica, which we don't want to do if someone is making invalid requests on data connection.

Reply:
That's why we are verifying whether request is having valid header i.e, msg.version (although this is also not a correct approach, since anyone can send the same version in the payload while making request, but it is still better approach IMO, otherwise we need to TLS server to validate the clients, which we want to avoid at this point of time)

Reply:
You have already read MagicVersion from the wire in func s.wire.Read() and now there is no version on wire to be read. Please correct me if I am wrong.

Reply:
@payes you're right. Have updated the PR